builtin/game/forceloading.lua

```diff
@@ -15,32 +16,54 @@ local function get_blockpos(pos)
 		z = math.floor(pos.z/BLOCKSIZE)}
 end
 
-function core.forceload_block(pos)
+function core.forceload_block(pos, transient)
 	local blockpos = get_blockpos(pos)
 	local hash = core.hash_node_position(blockpos)
-	if blocks_forceloaded[hash] ~= nil then
-		blocks_forceloaded[hash] = blocks_forceloaded[hash] + 1
+	local relevant_table
+	local other_table
+	if transient then
+		relevant_table = blocks_temploaded
+		other_table = blocks_forceloaded
+	else
+		relevant_table = blocks_forceloaded
+		other_table = blocks_temploaded
+	end
+	if relevant_table[hash] ~= nil then
+		relevant_table[hash] = relevant_table[hash] + 1
 		return true
+	elseif other_table[hash] ~= nil then
+		relevant_table[hash] = 1
 	else
 		if total_forceloaded >= (tonumber(core.setting_get("max_forceloaded_blocks")) or 16) then
 			return false
 		end
 		total_forceloaded = total_forceloaded+1
-		blocks_forceloaded[hash] = 1
+		relevant_table[hash] = 1
 		forceload_block(blockpos)
 		return true
 	end
 end
 
-function core.forceload_free_block(pos)
+function core.forceload_free_block(pos, transient)
 	local blockpos = get_blockpos(pos)
 	local hash = core.hash_node_position(blockpos)
-	if blocks_forceloaded[hash] == nil then return end
-	if blocks_forceloaded[hash] > 1 then
-		blocks_forceloaded[hash] = blocks_forceloaded[hash] - 1
+	local relevant_table
+	local other_table
+	if transient then
```

Comment:
Do you mean to make a function that takes a boolean and returns relevant_table, other_table?